index.js

```diff
@@ -11,68 +11,145 @@ module.exports = {
     'plugin:vue/recommended'
   ],
   plugins: [
-    'vue',
-    'jest'
+    'jest',
+    'unicorn',
+    'vue'
   ],
   settings: {
     'import/resolver': {
       node: { extensions: ['.js', '.mjs'] }
     }
   },
   rules: {
+
+    /**********************/
+    /* General Code Rules */
+    /**********************/
+
     // Enforce import order
-    'import/order': 2,
+    'import/order': 'error',
 
     // Imports should come first
-    'import/first': 2,
+    'import/first': 'error',
 
     // Other import rules
-    'import/no-mutable-exports': 2,
+    'import/no-mutable-exports': 'error',
 
     // Allow unresolved imports
-    'import/no-unresolved': 0,
+    'import/no-unresolved': 'off',
 
     // Allow paren-less arrow functions only when there's no braces
-    'arrow-parens': [2, 'as-needed', { requireForBlockBody: true }],
+    'arrow-parens': ['error', 'as-needed', { requireForBlockBody: true }],
 
     // Allow async-await
-    'generator-star-spacing': 0,
+    'generator-star-spacing': 'off',
 
     // Allow debugger during development
-    'no-debugger': process.env.NODE_ENV === 'production' ? 2 : 0,
-    'no-console': process.env.NODE_ENV === 'production' ? 2 : 0,
+    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',
+    'no-console': process.env.NODE_ENV === 'production' ? 'error' : 'off',
 
     // Prefer const over let
-    'prefer-const': [2, {
+    'prefer-const': ['error', {
       'destructuring': 'any',
       'ignoreReadBeforeAssign': false
     }],
 
     // No single if in an "else" block
-    'no-lonely-if': 2,
+    'no-lonely-if': 'error',
 
     // Force curly braces for control flow,
     // including if blocks with a single statement
-    curly: [2, 'all'],
+    curly: ['error', 'all'],
 
     // No async function without await
-    'require-await': 2,
+    'require-await': 'error',
 
     // Force dot notation when possible
-    'dot-notation': 2,
+    'dot-notation': 'error',
 
-    'no-var': 2,
+    'no-var': 'error',
 
     // Force object shorthand where possible
-    'object-shorthand': 2,
+    'object-shorthand': 'error',
 
     // No useless destructuring/importing/exporting renames
-    'no-useless-rename': 2,
+    'no-useless-rename': 'error',
+
+    /**********************/
+    /*   Unicorn Rules    */
+    /**********************/
+
+    // Pass error message when throwing errors
+    'unicorn/error-message': 'error',
+
+    // Uppercase regex escapes
+    'unicorn/escape-case': 'error',
+
+    // Use "new" for most builtins (not String, Number, ..., though)
+    'unicorn/new-for-builtins': 'error',
+
+    // Array.isArray instead of instanceof
+    'unicorn/no-array-instanceof': 'error',
+
+    // for-of instead traditional for loop
+    'unicorn/no-for-loop': 'error',
```

Comment:
A language feature is not really `traditional` because we have another type of loop. Someone should be really a newbie to use `index`/`array[index]` and it is obvious during code review. `for of` is really cool but not necessarily _better_ or an _alternative_ to `for`

Reply:
It's not as prone to off-by-one errors as a common for loop. IMO that's already "better" 🙈

Reply:
I don't understand. Better because why? :)) Just because it is more common it is adviced to already stick with?

Reply:
Not sure if I can follow that argumentation :thinking: 

I think:

---

#### Rules generally

1. ESLint rules are meant to cover the most common and most frequent use cases. There are exceptions for almost every rule (e.g. unsafe regex) but that's no reason to *not* include a rule.
1. Rules can *and should* be disabled for some exception cases
1. Rules can be disabled if people don't agree, similar to how nuxt.config.js defaults can be overridden

---

####  For loop rule 

1. "Traditional for loops" as in `for (let i = 0; i <= someArray.length; i++){ /***/ }` are prone to off-by-one errors, a very common error actually (as shown in the code snipped :yum:)
1. `for-of` loops are, compared to "traditional for loops" easier to read for the majority of the programmers (bold assumption here)
1. Enforcing a *consistent* code style is important for the code quality
1. If people don't like for-of loops or prefer traditional ones, they can remove/disable the rule, or use their own presets in case they don't agree with most of the rules in the nuxt preset

Reply:
> Rules can be disabled if people don't agree, similar to how nuxt.config.js defaults can be overridden

OK. And we are talking about default here not possiblity to disable. It's not even clear for me why in scope of nuxt lint rules, we should set a personal preference like this.

Reply:
Yes, we are trying to set defaults here which should help to write high-quality, consistent, readable and maintainable code. Of course, there is much more needed than a few rules but they help :relaxed: 

Also, please check https://github.com/nuxt/eslint-config/pull/35#issuecomment-486669222
We should agree on one "direction" which we are going.

Reply:
But the point is that `for of` and `for` are not alternative to each other to choose one. `for of` triggers the iterator protocol. `for` usage is rare, but valid.

Reply:
Can't we have different modes actually with this eslint-config? Like vue have: https://eslint.vuejs.org/rules/

So we could have `base` and `recommended`?
Also, it seems that we "force" the `plugin:vue/recommended` if I am correct

Reply:
@Atinux We could do that and enforce the base config only. But this is only worth it if we have lots of rules that don't/wouldn't fit in a base config. Or if we go with `base` for a more relaxed config and `recommended` with a more strict one.

Reply:
@Atinux please see conversation above. This something IMO not a `recommended` rule to advice!